docs/dom-testing-library/api-helpers.md

```diff
@@ -148,6 +148,30 @@ cy.get('form').within(() => {
 
 <!--END_DOCUSAURUS_CODE_TABS-->
 
+## `getRoles`
+
+This function allows iteration over the implicit aria roles represented in a
+given tree of DOM nodes. It returns an object indexed by role name with each
+value being an array of elements which have that implicit aria role.
+
+```javascript
+import { getRoles } from 'dom-testing-library'
+
+const nav = document.createElement('nav')
+nav.innerHTML = `
+<ul name="menu">
```

Comment:
Since `name` isn't really a valid attribute for ul/li elements, I'd suggest removing it, so there's also less code to read in the example.

What do you think about it?

Reply:
Oops, I didn't realize name isn't valid for ul/li.  

I think we should have some attribute in there to illustrate how it prints out the attributes.  Thoughts?

Assuming the above, I'll probably change name="menu" to type="circle" on the ul and give the li elements a 'value' attribute in place of 'name'.

Reply:
Changed here: [da985145](https://github.com/testing-library/testing-library-docs/pull/155/commits/da9851456fff50320a83c7c268e43f722264bda4)

Reply:
Didn't know that until I checked right now, but `type` attr on `<ul>` elements is discouraged: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/ul#attr-type

Also, looks like `value` on `<ul>` list items doesn't make much sense: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/li#attr-value

Reply:
Hmm.. swing and a miss, haha  

nav->ul->li was nice because it is three distinct roles in a real world kind of situation.  

Maybe it doesn't matter so much if the example illustrates how the output prints attributes, so there's always the option of getting rid of the attributes in the example (as your initial suggestion).  

Though I guess we could always add class and/or id attributes to illustrate a more real-world output.

I'm gonna push one more commit this evening.

Reply:
;) no prob.

> Maybe it doesn't matter so much if the example illustrates how the output prints attributes, so there's always the option of getting rid of the attributes in the example.

This still makes sense to me. The main goal of the example is to illustrate what a grouping-by-role looks like, not the attribute themselves.

Thanks for your work!

Reply:
Ok, I've pulled the attributes off the elements completely.  I agree that it's the correct call.  Sorry for the delay, distracted by work/holiday.